### PR TITLE
Feat: Enforce mandatory fuel code validation for schedules - 1429

### LINKED ISSUE #1429: LCFS - Schedules, Fuel code validation
**Description:**  
For the 4 schedules input forms:
1. Fuel supply
2. Export fuel
3. Allocation agreement
4. Other use

If the option 'Fuel code - section 19 (b) (i)' is selected under 'Determining carbon intensity', the 'Fuel code' cell must become mandatory. It should not pass validation or allow saving until a valid fuel code is selected. If only one fuel code exists, it should auto-populate. If multiple fuel codes exist, no default value should be set; the user must explicitly select one.

**Wireframe:**  
Not applicable.

**Purpose and benefit to user:**  
This ensures accurate and complete data entry for determining carbon intensity, reducing errors and enhancing compliance with validation rules.

**Acceptance Criteria:**  
- [ ] When 'Fuel code - section 19 (b) (i)' is selected in the 'Determining carbon intensity' dropdown:
  - [ ] The 'Fuel code' cell becomes mandatory.
  - [ ] Saving the form without selecting a 'Fuel code' triggers a validation error.
- [ ] If only one fuel code exists:
  - [ ] The 'Fuel code' cell auto-populates with that code.
- [ ] If multiple fuel codes exist:
  - [ ] No default fuel code is pre-selected.
  - [ ] The user is required to manually select a fuel code before saving.

**Development Checklist:**  
- [ ] Implement logic to make the 'Fuel code' cell mandatory when 'Fuel code - section 19 (b) (i)' is selected.
- [ ] Configure auto-population of 'Fuel code' when only one option exists.
- [ ] Prevent auto-selection when multiple fuel codes are available; enforce manual selection.
- [ ] Add validation error messaging for unselected mandatory 'Fuel code'.
- [ ] Apply these rules to all 4 schedules: Fuel supply, Export fuel, Allocation agreement, and Other use input forms.
- [ ] Test scenarios with single and multiple fuel codes to ensure correct behavior.

**Notes:**  
- Confirm the list of fuel codes and their availability across different schedules.
- Ensure error messages are user-friendly and clearly indicate the requirement to select a 'Fuel code'.

### PR DESCRIPTION
**Pull request by hamed-valiollahi**
This PR implements logic to enforce mandatory fuel code selection in schedules when "Fuel code - section 19 (b) (i)" is chosen.

Closes #1429

### CODE REVIEW

File: backend/lcfs/web/application.py [hunk 1]
```diff
@@ -1,4 +1,6 @@
 import logging
+import os
+import debugpy
```
Comment (areyeslo) [inline backend/lcfs/web/application.py:3]: Do we need these new imports?

**Review by areyeslo — CHANGES_REQUESTED** [review]
Looks great! Tested and works great!. Just wondering if we need those imports?

**Comment by hamed-valiollahi** [comment]
> Looks great! Tested and works great!. Just wondering if we need those imports?

Thanks for the code review. We need these imports for backend debugging purposes. It looks like they were accidentally removed. They enable remote debugging in development if needed. Please refer to line 111 of this file for more details.

**Review by areyeslo — APPROVED** [review]
LGTM

**Pull request merged**